util/predicates/configmap_predicates.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ConfigMapCreate returns a predicate that returns true for a ConfigMap create event
+func ConfigMapCreateUpdate(logger logr.Logger) predicate.Funcs {
```

Comment:
It looks like this can be made to handle any type by using `e.Meta.GetNamespace()` and `e.Meta.GetName()` rather than duplicating these predicates for different types and doing the type conversions against `runtime.Object`

Reply:
Without the conversion, how will we know if the create if for ConfigMaps/Secrets?

Reply:
I'm not sure that we really care for the purpose of the predicate, since the type is already constrained in the Watch we are defining, and we can infer it in the mapping function (where it likely matters more)

Reply:
Got it, will see if it works as expected.

Btw, the current implementation is buggy, mapping function is triggered constantly, I could not find what is wrong. Any ideas?

Reply:
I'm not seeing anything that is jumping out immediately

Reply:
Try simplifying this to return true for create & update, and false for delete (you don't need to assert that it's a ConfigMap or Secret).

Reply:
Did simplify prediates as suggested, but did not combine them as need to check Secret type in Predicates.